### PR TITLE
Us050 manually change the response status

### PR DESCRIPTION
**Pull request by benjefferies**
As a internal user (Survey administrator?)
I need to change the case response status to completed by phone for an
RU for a specific CE for a specific survey
So that the respondent can complete the survey over the phone and they do not receive reminders
    
* Add change link onto reporting-unit.html
* Add change-response-status.html for changing status

Testing
1. Run up ras/rm
1. Run response-operations-ui on branch `us050-manually-change-the-response-status`
1. Run ras-backstage on branch `us050a-change-response-status`
1. Build repo `rm-common-service` branch `us050a-change-response-status-completedbyphone`
1. Build rm-casesvc-api on branch `us050a-change-response-status`
1. Run rm-case-service on branch `us050a-change-response-status-completedbyphone`
1. Load data with `make setup`
1. Open http://localhost:8085/reporting-units/49900000001
1. Click change
1. See updated status

[trello](https://trello.com/c/q5skwt1o/91-us050a-int-manually-change-the-response-status-for-a-given-ru-survey-ce-to-completed-by-phone-xl)

### CODE REVIEW

File: response_operations_ui/controllers/case_controller.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def get_available_case_group_statuses(short_name, period, ru_ref):
+    logger.debug('Retrieving case group status', short_name=short_name, period=period, ru_ref=ru_ref)
+    url = f'{app.config["BACKSTAGE_API_URL"]}/v1/case/status/{short_name}/{period}/{ru_ref}'
+    response = requests.get(url)
+    if response.status_code != 200:
+        raise ApiError(response)
```
Comment (andrewmil) [inline response_operations_ui/controllers/case_controller.py]: Untested error

File: response_operations_ui/controllers/case_controller.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    logger.debug('Successfully retrieved case group status', short_name=short_name, period=period, ru_ref=ru_ref)
+    return response.json()
+
+
+def update_case_group_statuses(short_name, period, ru_ref, event):
+    logger.debug('Updating case group status', short_name=short_name, period=period, ru_ref=ru_ref)
+    url = f'{app.config["BACKSTAGE_API_URL"]}/v1/case/status/{short_name}/{period}/{ru_ref}'
+    response = requests.post(url, json={'event': event})
+    if response.status_code != 200:
+        raise ApiError(response)
```
Comment (andrewmil) [inline response_operations_ui/controllers/case_controller.py]: Untested error

File: response_operations_ui/templates/change-response-status.html [hunk 3]
```diff
@@ -0,0 +1,106 @@
+{% extends "layouts/base.html" %}
```
Comment (andrewmil) [inline response_operations_ui/templates/change-response-status.html]: In general this html seems a bit spaced out, might be good to make it a little bit more compact

File: response_operations_ui/views/case.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+def update_response_status(ru_ref):
+    short_name = request.args.get('survey')
+    collection_exercise_period = request.args.get('period')
+    form = ChangeGroupStatusForm(request.form)
+    if form.event:
+        case_controller.update_case_group_statuses(short_name, collection_exercise_period, ru_ref, form.event.data)
+        return redirect(url_for('reporting_unit_bp.view_reporting_unit', ru_ref=ru_ref,
+                        survey=short_name, period=collection_exercise_period))
+
+    statuses = case_controller.get_available_case_group_statuses(short_name, collection_exercise_period, ru_ref)
```
Comment (andrewmil) [inline response_operations_ui/views/case.py]: Everything below here is not covered in the unit tests

File: response_operations_ui/views/case.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+from werkzeug.utils import redirect
+
+from response_operations_ui.common.mappers import map_ce_response_status
+from response_operations_ui.controllers import case_controller
+from response_operations_ui.forms import ChangeGroupStatusForm
+
+case_bp = Blueprint('case_bp', __name__, static_folder='static', template_folder='templates')
+
+
+@case_bp.route('/<ru_ref>/change-response-status', methods=['GET'])
```
Comment (andrewmil) [inline response_operations_ui/views/case.py]: Is this the correct url for this? Best to consult with Nic on what he'd like this to be

File: tests/views/test_case.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    def test_get_available_statuses_fail(self, mock_request):
+        short_name = 'MYSURVEY'
+        period = '221_202018'
+        ru_ref = '19000001'
+        mock_request.get(f'{app.config["BACKSTAGE_API_URL"]}/v1/case/statuses/{short_name}/{period}/{ru_ref}',
+                         status_code=500)
+
+        response = self.app.get(f'/case/{ru_ref}/change-response-status?survey={short_name}&period={period}')
+
+        self.assertEqual(response.status_code, 302)
```
Comment (andrewmil) [inline tests/views/test_case.py]: It's probably better to add `follow_redirects=True` to the get request and do an assert on the fail page

File: response_operations_ui/views/case.py (hunk 5 above)
Comment (benjefferies) [reply response_operations_ui/views/case.py]: I'll double check but in the prototype it looks this
image
I think the query params are just for the prototype as we can look it up

File: response_operations_ui/templates/change-response-status.html (hunk 3 above)
Comment (benjefferies) [reply response_operations_ui/templates/change-response-status.html]: Yeah sure it was just a lazy code and paste :)

**Comment by tej99** [comment]
Scenario "Able to view all collection exercises associated to the displayed RU Ref" will fail in the acceptance test when this is merged to master. This is because  row.find_by_name('tbl-ce-status').value will now capture 'Not started  Change' instead of 'Not started'.

**Comment by liamtoozer** [comment]
@tej99 Good spot. I've amended that test [here](https://github.com/ONSdigital/rasrm-acceptance-tests/pull/63/commits/428e2ed0b3b93638b1590c1a248b7e908fad431d), so it should pass again hopefully.

**Comment by madeye-matt** [comment]
I've checked this out and tested it too - it all looks fine (can't comment on the perfection of the python though). 
@benjefferies  I've also just merged in the latest from master so it should be good to go

**Pull request merged**